### PR TITLE
Metallicity evolution plots

### PR DESCRIPTION
**Pull request by MatthieuSchaller**
Extending the fun a bit more

### CODE REVIEW

File: eagle-xl/scripts/bh_metallicity_evolution.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    redshift = data.z
+    bh_Z_mass = data.bh_z_mass
+    bh_Z_mass_density = bh_Z_mass / box_volume
+
+    # High z-order as we always want these to be on top of the observations
+    simulation_lines.append(ax.plot(scale_factor, bh_Z_mass_density, zorder=10000)[0])
+    simulation_labels.append(name)
+
+ax.set_xlabel("Redshift $z$")
+ax.set_ylabel(r"Metal Mass in Black Holes $\rho_{\rm BH,Z}$ [M$_\odot$ Mpc$^{-3}$]")
```
Comment (JBorrow) [inline eagle-xl/scripts/bh_metallicity_evolution.py:62]: This is not necessarily true. Please use `bh_Z_mass_density.to("Solar_Mass / Mpc**3")`.

Comment (JBorrow) [reply eagle-xl/scripts/bh_metallicity_evolution.py:62]: In fact it won't be - the value produced will be 1e10 M_\odot Mpc^-3

File: eagle-xl/scripts/metallicity_evolution.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            star_Z_mass_density,
+            linestyle="dashed",
+            color=f"C{color}",
+            zorder=10000,
+        )[0]
+    )
+    simulation_labels.append(name)
+
+ax.set_xlabel("Redshift $z$")
+ax.set_ylabel(r"Metal Mass $\rho_{\rm Z}$ [M$_\odot$ Mpc$^{-3}$]")
```
Comment (JBorrow) [inline eagle-xl/scripts/metallicity_evolution.py:81]: Again requires use of the unit conversions. Alternatively, you can use the `unyt.matplotlib_support` environment (see https://unyt.readthedocs.io/en/stable/usage.html#plotting-with-matplotlib)

**Review by JBorrow — COMMENTED** [review]
Needs to use the unit conversions. Are these missing from the existing plots too?

**Comment by MatthieuSchaller** [comment]
I thought your reading script was handling the unit conversions correctly. The numbers that come out make sense currently, so I don't think anything needs changing here.

**Comment by JBorrow** [comment]
What I mean is that it's by chance. They are symbolic units - they may be in CGS, depending on the output. If you force a conversion then we _know_ it is in those units. At the moment I guess it's by chance correct but that may not always be the case.

**Comment by MatthieuSchaller** [comment]
Ok but then I don't understand what you want me to do here.

File: eagle-xl/scripts/bh_metallicity_evolution.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    snapshot = load(snapshot_filename)
+    boxsize = snapshot.metadata.boxsize
+    box_volume = boxsize[0] * boxsize[1] * boxsize[2]
+
+    # a, Redshift, SFR
+    scale_factor = data.a
+    redshift = data.z
+    bh_Z_mass = data.bh_z_mass
+    bh_Z_mass_density = bh_Z_mass / box_volume
```
Comment (JBorrow) [inline eagle-xl/scripts/bh_metallicity_evolution.py:55]: If here you do `bh_Z_mass_density = (bh_Z_mass / box_volume).to("Solar_Mass / Mpc**3")` you will always ensure that the value that you plot has those units, if that makes sense.

**Comment by JBorrow** [comment]
Great, thanks. I'll take another pass at this tomorrow before merging but it looks good to me.

One case where this would have definitely gone wrong is simulations performed with kpc as the internal length unit. Not that we're doing that right now, but you never know...

**Comment by MatthieuSchaller** [comment]
Yes, makes sense now.

**Comment by JBorrow** [comment]
Thanks!

**Pull request merged**